petab/lint.py

```diff
@@ -20,20 +21,28 @@ def condition_table_is_parameter_free(condition_df):
     constant_parameters = list(set(condition_df.columns.values.tolist()) - {'conditionId', 'conditionName'})
 
     for column in constant_parameters:
-        if np.any(np.invert(condition_df.loc[:, column].apply(isinstance, args=(float,)))):
+        if np.any(np.invert(condition_df.loc[:, column].apply(isinstance, args=(numbers.Number,)))):
             return False
     return True
 
 
 def measurement_table_has_timepoint_specific_mappings(measurement_df):
     """Are there time-point or replicate specific parameter assignments in the measurement table"""
 
-    measurement_df.loc[measurement_df.noiseParameters.apply(isinstance, args=(float,)), 'noiseParameters'] = np.nan
+    measurement_df.loc[
+        measurement_df.noiseParameters.apply(isinstance, args=(numbers.Number,)), 'noiseParameters'] = np.nan
```

Comment:
I guess we just had the issue with nans here, i.e. grouping difficult when all are nans? did this work now?

Reply:
Changed.